### ISSUE TITLE
False positive `lower_acl_than_parent` in public extension

### ISSUE THREAD

**Issue by dzungaria** [realm/SwiftLint#4564]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`lower_acl_than_parent` rule reports a false positive for members of types defined in `public extension`.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint --config swiftlint.yml lower_acl_than_parent.swift
Linting Swift files at paths lower_acl_than_parent.swift
Linting 'lower_acl_than_parent.swift' (1/1)
lower_acl_than_parent.swift:5:9: warning: Lower ACL than parent Violation: Ensure declarations have a lower access control level than their enclosing parent (lower_acl_than_parent)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
  0.50.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
  Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - lower_acl_than_parent
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  No
* Which Xcode version are you using (check `xcodebuild -version`)?
  Xcode 14.1
  Build version 14B47b
* Do you have a sample that shows the issue?

```swift
public enum Namespace {}

public extension Namespace {
    struct Box {
        public let value: String // This line triggers a false positive
    }
}
```

**Comment by leohidalgo** [comment]
+1

**Comment by jpsim** [comment]
Fixed in #4575 by @HeEAaD
